lib/fog/core/services_mixin.rb

```diff
@@ -24,16 +24,22 @@ def providers
 
     private
 
+    # This method should be removed once all providers are extracted.
+    # Bundler will correctly require all dependencies automatically and thus
+    # fog-core wont need to know any specifics providers. Each provider will
+    # have to load its dependencies.
     def require_service_provider_library(service, provider)
       require "fog/#{provider}/#{service}"
     rescue LoadError  # Try to require the service provider in an alternate location
+      Fog::Logger.deprecation('Prefer to define all Services within the Provider Namespace instead!')
```

Comment:
I think maybe a more specific error would be helpful here. Like `Unable to load "#{fog/#{provider}/#{service}"`. What do you think?

Reply:
Hmmm. Good idea. How to announce the deprecation as well?